presto-parser/src/main/java/com/facebook/presto/sql/tree/DynamicFilterExpression.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class DynamicFilterExpression
+        extends Expression
+{
+    private final String sourceId;
+    private final ComparisonExpressionType type;
+    private final SymbolReference probeSymbolReference;
+    private final String dfSymbol;
```

Comment:
I would rather have an `Assignments` here, e.g: `dynamic_filter_sym -> probe_expression`. 

This is because with many join conditions, e.g: `probe_sym1=build_sym1 AND probe_sym2=build_sym2` you can implement dynamic filtering efficiently as hash filtering on column tuples instead of hash filtering on each column one by one.

Reply:
Assignments not accessible here.